drivers/SmartThings/zigbee-thermostat/src/vimar/init.lua

```diff
@@ -0,0 +1,213 @@
+    end)
+  end
+end
+
+-- NOTE: unused attributes are not refreshed
+local vimar_thermostat_do_refresh = function(self, device)
+  local attributes = {
+    Thermostat.attributes.LocalTemperature,
+    Thermostat.attributes.ControlSequenceOfOperation,
+    Thermostat.attributes.ThermostatRunningState,
+    Thermostat.attributes.SystemMode
+  }
+
+  local vimar_thermostat_profile = device:get_field(VIMAR_CURRENT_PROFILE)
+  if vimar_thermostat_profile == ThermostatMode.thermostatMode.heat.NAME then
+    attributes[#attributes + 1] = Thermostat.attributes.OccupiedHeatingSetpoint
+  elseif vimar_thermostat_profile == ThermostatMode.thermostatMode.cool.NAME then
+    attributes[#attributes + 1] = Thermostat.attributes.OccupiedCoolingSetpoint
+  end
+
+  for _, attribute in pairs(attributes) do
+    device:send(attribute:read(device))
+  end
+end
+
+-- NOTE: Whenever the physical button for the current mode is pressed on the device, this function changes the device profile.
+--       If the Thermostat is OFF, no profile change is required.
+local vimar_thermostat_mode_handler = function(driver, device, thermostat_mode)
+  local mode = VIMAR_THERMOSTAT_MODE_MAP[thermostat_mode.value].NAME
+  -- If is a known supported mode, then apply the change
+  if VIMAR_THERMOSTAT_MODE_MAP[thermostat_mode.value] then
+    local vimar_thermostat_profile = device:get_field(VIMAR_CURRENT_PROFILE)
+    -- HEAT: if the previous mode was cool, update profile
+    if mode == ThermostatMode.thermostatMode.heat.NAME then
+      if mode ~= vimar_thermostat_profile then
+        device:try_update_metadata({ profile = VIMAR_THERMOSTAT_HEATING_PROFILE })
+        device:set_field(VIMAR_CURRENT_PROFILE, mode)
+        device.thread:call_with_delay(2, function(d)
+          vimar_thermostat_do_refresh(driver, device)
+        end)
+      end
+      -- COOL: if the previous mode was heat, update profile
+    elseif mode == ThermostatMode.thermostatMode.cool.NAME then
+      if mode ~= vimar_thermostat_profile then
+        device:try_update_metadata({ profile = VIMAR_THERMOSTAT_COOLING_PROFILE })
+        device:set_field(VIMAR_CURRENT_PROFILE, mode)
+        device.thread:call_with_delay(2, function(d)
+          vimar_thermostat_do_refresh(driver, device)
+        end)
+      end
+    end
+    device:emit_event(VIMAR_THERMOSTAT_MODE_MAP[thermostat_mode.value]())
+  end
+end
+
+-- NOTE: override default thermostat mode map; logic is the same from the original driver
+local vimar_set_thermostat_mode = function(driver, device, command)
+  for zigbee_attr_val, st_cap_val in pairs(VIMAR_THERMOSTAT_MODE_MAP) do
+    if command.args.mode == st_cap_val.NAME then
+      device:send_to_component(command.component, Thermostat.attributes.SystemMode:write(device, zigbee_attr_val))
+      device.thread:call_with_delay(1, function(d)
+        device:send_to_component(command.component, Thermostat.attributes.SystemMode:read(device))
+      end)
+      break
+    end
+  end
+end
```

Comment:
Because you're setting your supported modes attribute, users shouldn't be able to set unsupported modes unless they're specifically trying to by going through the API. If that's something you're worried about you can leave this in, but you could also take it out and be fine.